sdk/nodejs/cmd/run/run.ts

```diff
@@ -235,7 +235,7 @@ export function run(
         const defaultMessage = err.stack || err.message || ("" + err);
 
         // First, log the error.
-        if (RunError.isInstance(err)) {
+        if (RunError.isInstance(err) || typeof err ==  typeof tsnode.TSError) {
```

Comment:
I'm not super familiar with TS, but shouldn't it be:
```suggestion
        if (RunError.isInstance(err) || tsnode.TSError.isInstance(err)) {
```

Reply:
https://github.com/pulumi/pulumi/blob/a92a005d6897506d3ea40307dfd37788c51a4dcf/sdk/nodejs/errors.ts#L38-L40

isInstance appears to be a specific method on the RunError class. I tried that one first :)